[1.4.1] 게임 오버 시 화면 조정

MtNight opened the pull request:
1. 게임 오버 시 clearGameScreen 함수로 화면을 지우지 않고 기존 화면을 남겨두도록 조건 추가
2. drawObjects 함수가 죽기 직전 상태를 보존하고, 머리를 빨간색으로 칠함.
3. 빨간색을 매크로 변수 QUIT_COLOR로 정의 후 적용

#39 이슈의 논의 과정을 토대로 변경되었습니다.

MtNight committed:
[MOD] Add condition in clearGameScreen call

I add a if-statement in clearGameScreen call part.
Now, Screen isn't cleared in game over situation.

MtNight committed:
[MOD] Modify drawObject function

I modify 'drawObject' function for game over situation.
It is included adding parameter and coloring snake's head.

MtNight committed:
[NEW] Add QUIT_COLOR macroValue

I add macro value 'QUIT_COLOR' to denoting game over.
It is implemented in drawObjects function.

mellime commented:
추가해주신 내용은 전부 확인했습니다!
다만, main.c 에서 quit 변수는 bool 타입인데, drawObject 함수에서는 형변환을 거쳐 int 타입으로 받으시는 건가요?

MtNight committed:
[MOD] Change type of drawObject's parameter

I change the data type of 'quit(drawObject's parameter)' value to 'bool' from 'int'.

MtNight commented:
@mellime 으어 왜 int로 했을까요
수정했습니다 확인 부탁드립니다...

mellime commented:
확인했습니다.
drawObject 함수의 조건은 if (quit) 로 그대로 두셔도 될 것 같긴 해요!

MtNight commented:
main.c의 충돌 함수들과 일관성을 위해서 그렇게 했습니다. 만약 변경한다면 나중에 그것들도 한번에 변경하는 것이 좋을 것 같습니다.

hongyj9823 commented:
#39 이슈 논의 과정 확인했습니다. 죽기 직전의 과정에 빨간색으로 바뀐 거 좋은 아이디어 같습니다. 확인 완료했습니다.
